core/lib/spree/permission_sets/user_management.rb

```diff
@@ -2,7 +2,16 @@ module Spree
   module PermissionSets
     class UserManagement < PermissionSets::Base
       def activate!
-        can :manage, Spree.user_class
+        can [:admin, :display, :create, :update], Spree.user_class
+
+        # due to how cancancan filters by associations,
+        # we have to define this twice, once for `accessible_by`
+        can :update_email, Spree.user_class, spree_roles: { id: nil }
+        # and once for `can?`
+        can :update_email, Spree.user_class do |user|
+          user.spree_roles.none?
```

Comment:
I'm pro restricting user role changes to just super users on the assumption that the average user does not have roles, and the assigning of roles to a user is at a level above regular user management. The permission set can be extended if users desire the ability, or want to change it, or an entirely new one could be created depending on how flexibile they need it to be. I think taking the safest approach here is probably the best approach.

:+1: